compiler-core/src/type_/expression.rs

```diff
@@ -528,13 +568,22 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
             implicit: true,
         });
 
-        let call = self.infer(UntypedExpr::Call {
-            location: SrcSpan::new(use_.location.start, sequence_location.end),
-            fun: call.function,
-            arguments: call.arguments,
-        })?;
+        let call = self.infer_call(
+            *call.function,
+            call.arguments,
+            call_location,
+            CallKind::Use {
+                assignments_location,
+                last_statement_location,
+            },
+        )?;
 
-        Ok(Statement::Expression(call))
+        Ok(Use {
```

Comment:
This has radically changed the meaning and data of `Use`- now the callback code is within the call in the `Use`. Can we document this somewhere? Or is there some way to make it clearer?

Reply:
Ok, I'm adding some comments!